src/Symfony/Component/EventDispatcher/DependencyInjection/RegisterListenersPass.php

```diff
@@ -96,7 +96,7 @@ public function process(ContainerBuilder $container)
                     $dispatcherDefinition = $container->getDefinition($event['dispatcher']);
                 }
 
-                $dispatcherDefinition->addMethodCall('addListener', [$event['event'], [new ServiceClosureArgument(new Reference($id)), $event['method']], $priority]);
+                $dispatcherDefinition->addMethodCall('addListener', [$event['event'], [new ServiceClosureArgument(new Reference($id), $container->getDefinition($id)->getClass()), $event['method']], $priority]);
```

Comment:
should be findDefinition() I think (same below)

Reply:
Done